include/swift/AST/Types.h

```diff
@@ -6033,6 +6092,19 @@ inline bool TypeBase::isTypeParameter() {
   return false;
 }
 
+inline bool TypeBase::isTypeParameterRootedOn(
+    const GenericTypeParamType *gp) {
+  Type t(this);
+
+  while (auto *const depMemTy = t->getAs<DependentMemberType>())
```

Comment:
Nit: Is this `const` misplaced?

Reply:
cpp "vars" that are never mutated are a pet peeve of mine, especially in large scopes. This one's a bit of an overkill though; I certainly do not mind making it a pointer to `const`.